src/goto-instrument/synthesizer/loop_invariant_synthesizer.cpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************\
+
+Module: Synthesize Loop Invariants
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+/// \file
+/// Synthesize Loop Invariants
+
+#include "loop_invariant_synthesizer.h"
+
+void enumerative_loop_invariant_synthesizert::init_candidates()
+{
+  for(auto &function_p : goto_model.goto_functions.function_map)
+  {
+    natural_loops_mutablet natural_loops(function_p.second.body);
+
+    // Initialize invairants for unannotated loops as true
+    for(unsigned int idx = 0; idx < natural_loops.loop_map.size(); idx++)
+    {
+      loop_idt new_id;
+      new_id.func_name = function_p.first;
+      new_id.loop_number = idx;
```

Comment:
So you are numbering via `natural_loops`, but there's also the loop numbers that `goto-instrument --show-loops` will produce. I am not sure these numbers will coincide. Was there a reason to go via `natural_loops` here?

Reply:
I don't have any specified reason to use ```natural_loops``` here. Yes, it would be better if we use the same ```loop_number``` as ```--show-loops``` show. I will modify it.